url-builder/src/test/java/com/palominolabs/http/url/UrlBuilderTest.java

```diff
@@ -411,6 +411,26 @@ public void testClearQueryAfterQueryAllowsQueryParam() throws CharacterCodingExc
         assertUrlEquals("http://host?foo=bar", ub.toUrlString());
     }
 
+    @Test
+    public void testToUrlMatchesToUrlString() throws MalformedURLException, CharacterCodingException {
+        UrlBuilder ub = forHost("http", "host")
+                .unstructuredQuery("foobar")
+                .clearQuery()
+                .queryParam("foo", "bar");
+        assertEquals(ub.toUrlString(), ub.toUrl().toString());
```

Comment:
let's make this a more full featured url with path and fragment too so we can catch a change that drops passing those to the URL ctor